src/runtime_src/core/tools/xbmgmt2/SubCmdExamine.cpp

```diff
@@ -129,94 +129,91 @@ SubCmdExamine::execute(const SubCmdOptions& _options) const
-        std::cout << boost::format("  [%s] : %s\n") % _dev.get<std::string>("bdf") % _dev.get<std::string>("vbnv");
-      }
-      std::cout << std::endl;
-      throw xrt_core::error(std::errc::operation_canceled);
+  // when json is specified, make sure an accompanying output file is also specified
+  if (!sFormat.empty() && sOutput.empty())
+    throw xrt_core::error("Please specify an output file to redirect the json to");
+
+  if(sFormat.empty())
+    sFormat = "json";
+
+  // Output Format
+  schemaVersion = Report::getSchemaDescription(sFormat).schemaVersion;
+  if (schemaVersion == Report::SchemaVersion::unknown) 
+    throw xrt_core::error((boost::format("Unknown output format: '%s'") % sFormat).str());
+
+  // Output file
+  if (!sOutput.empty() && boost::filesystem::exists(sOutput) && !XBU::getForce()) 
+      throw xrt_core::error((boost::format("Output file already exists: '%s'") % sOutput).str());
+
+  // Collect all of the devices of interest
+  std::set<std::string> deviceNames;
+  for (const auto & deviceName : devices) 
+    deviceNames.insert(boost::algorithm::to_lower_copy(deviceName));
+
+  XBU::collect_devices(deviceNames, false /*inUserDomain*/, deviceCollection);
+
+  // enforce 1 device specification if multiple reports are requested
+  if(deviceCollection.size() > 1 && (reportsToProcess.size() > 1 || reportNames.front().compare("host") != 0)) {
+    std::cerr << "\nERROR: Examining multiple devices is not supported. Please specify a single device using --device option\n\n";
+    std::cout << "List of available devices:" << std::endl;
+    boost::property_tree::ptree available_devices = XBU::get_available_devices(false);
+    for(auto& kd : available_devices) {
+      boost::property_tree::ptree& _dev = kd.second;
+      std::cout << boost::format("  [%s] : %s\n") % _dev.get<std::string>("bdf") % _dev.get<std::string>("vbnv");
     }
-  } catch (const xrt_core::error& e) {
-    // Catch only the exceptions that we have generated earlier
-    std::cerr << boost::format("ERROR: %s\n") % e.what();
-    printHelp(commonOptions, hiddenOptions);
+    std::cout << std::endl;
     throw xrt_core::error(std::errc::operation_canceled);
   }
-  catch (const std::runtime_error& e) {
-    // Catch only the exceptions that we have generated earlier
-    std::cerr << boost::format("ERROR: %s\n") % e.what();
-    throw xrt_core::error(std::errc::operation_canceled);
+
+  bool is_report_output_valid = true;
+  // DRC check on devices and reports
+  if (deviceCollection.empty()) {
+    std::vector<std::string> missingReports;
+    for (const auto & report : reportsToProcess) {
+      if (report->isDeviceRequired())
+        missingReports.push_back(report->getReportName());
+    }
+
+    if (!missingReports.empty()) {
+      is_report_output_valid = false;
+      // Print error message
+      std::cerr << boost::format("Error: The following report(s) require specifying a device using the --device option:\n");
```

Comment:
I updated the test cases above to show a return code of non-zero for reports that fail to generate. There is no throw here because we need to account for the 'host' report. The throw occurs at the end of the function on line 235. It uses the same mechanism to that detects a report that failed to generate.